component/authentication-endpoint/src/main/webapp/smsotpError.jsp

```diff
@@ -50,6 +50,8 @@
                     errorMessage = "The code entered is incorrect. Authentication Failed!";
                 } else if (errorMessage.equalsIgnoreCase("smsotp.disable")) {
                     errorMessage = "Enable the SMS OTP in your Profile. Cannot proceed further without SMS OTP authentication.";
+                } else if (errorMessage.equalsIgnoreCase("token.expired")) {
+                    errorMessage = "The code entered is expired. Authentication Failed!";
```

Comment:
why did you use different error message for the same condition? 
line 49 : The code entered is expired. Click Resend Code to continue

Reply:
That was in smsotp.jsp file and where user able to retry the flow with same error message  and this one is used in smsotpError.jsp page where the only the error message shown if the retry is disabled.